### ISSUE TITLE
[youtube] "Video unavailable" with cookies - YouTube blocking accounts from web

### ISSUE THREAD

**Issue by kuylar** [yt-dlp/yt-dlp#10085]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Provide a description that is worded well enough to be understood

Recently, YouTube has started blocking accounts that are used a lot with yt-dlp from loading videos through the /player endpoint. Blocked accounts will get "Video unavailable. This content isn’t available." for all videos.

Blocked accounts can't load videos in the website, but the Android app works fine (same account and IP were used for the following screenshots)
| WEB client | Latest stock YouTube client on Android |
| --- | --- |
| ![Website, showing the same error as yt-dlp](https://github.com/yt-dlp/yt-dlp/assets/52961639/39af47f2-654d-4a45-b7be-9f6de6e1062e) | ![Android client, loading the same account without issues](https://github.com/yt-dlp/yt-dlp/assets/52961639/83910562-ba50-4011-a8a4-dbea12153c24) |

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['--cookies-from-browser', 'firefox', '-vU', 'dQw4w9WgXcQ']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.05.30.232720 from yt-dlp/yt-dlp-nightly-builds (linux_exe)
[debug] Python 3.11.9 (CPython x86_64 64bit) - Linux-6.8.8-arch1-1-x86_64-with (OpenSSL 3.1.4 24 Oct 2023)
[debug] exe versions: ffmpeg 6.1.1 (setts), ffprobe 6.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.02.02, mutagen-1.47.0, requests-2.32.3, secretstorage-3.3.3, sqlite3-3.44.2, urllib3-2.2.1, websockets-12.0
[debug] Proxy map: {}
Extracting cookies from firefox
[debug] Extracting cookies from: "/home/kuylar/.mozilla/firefox/mtlek44v.default-release-1708816247016/cookies.sqlite"
Extracted 2968 cookies from firefox
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1820 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.05.30.232720 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.05.30.232720 from yt-dlp/yt-dlp-nightly-builds)
[youtube] Extracting URL: dQw4w9WgXcQ
[youtube] dQw4w9WgXcQ: Downloading webpage
[debug] [youtube] Extracted SAPISID cookie
[youtube] dQw4w9WgXcQ: Downloading ios player API JSON
ERROR: [youtube] dQw4w9WgXcQ: Video unavailable. This content isn’t available.
  File "yt_dlp/extractor/common.py", line 734, in extract
  File "yt_dlp/extractor/youtube.py", line 4248, in _real_extract
  File "yt_dlp/extractor/common.py", line 1257, in raise_no_formats
```

**Comment by bashonly** [comment]
Related: #10046

**Comment by bashonly** [comment]
As reported by a user on discord whose account is also affected, video playback works via the iOS client as well. So `--extractor-arg "youtube:player_client=ios"` should be able to workaround the issue (if you need to keep passing cookies)

**Comment by triplesixman** [comment]
> As reported by a user on discord whose account is also affected, video playback works via the iOS client as well. So `--extractor-arg "youtube:player_client=ios"` should be able to workaround the issue (if you need to keep passing cookies)

Interesting. Are there any specifics to using the iOS client? What video formats are available, for example?

**Comment by cheese529** [comment]
> What video formats are available, for example?

you would get way less formats available that's for sure, I am hoping we can find a solution for this because we already lost the android client and having both web and iOS was a must for me. wishing the devs working on this the best of luck.

**Comment by bashonly** [comment]
Fewer formats, for sure, but not "way" fewer. Many of the iOS client formats are just HLS variants of the WEB formats (exact same quality but different protocol and format id).

**Comment by cheese529** [comment]
I see, but from what I gathered we'd be losing OPUS audio which was used most commonly when with youtube videos is that correct?

**Comment by bashonly** [comment]
> osing OPUS audio

Yeah, that's the most notable loss. Possibly format 141 (high quality aac audio) too.

**Comment by triplesixman** [comment]
I did a test with an account that was banned using the web client, I put “--extractor-args youtube:player_client=ios” and despite that I'm still banned.

Am I missing something? I'm using the same cookie I was banned with on the web client, I've just added the --extractor-args parameter

**Comment by bashonly** [comment]
Correction: the workaround is `--extractor-args "youtube:player_client=ios;player_skip=webpage"`if you need to continue passing cookies

**Comment by triplesixman** [comment]
It seems that account restrictions even occur with iOS player. So for the moment I'm not passing any more cookies (but I'm expecting new problems).

**Comment by triplesixman** [comment]
My IP is now limited to 40 kb/s (ios client, no cookie).

If anyone has a solution, I'm interested (there's no question of using a proxy, since all of them are or will be affected by this problem). Personally, I have no idea how to get around this. 

It looks like youtube has won this battle.

**Comment by cheese529** [comment]
> Yeah, that's the most notable loss. Possibly format 141 (high quality aac audio) too.

Yea unfortunately, do you think we would ever find a solution to this? Also are you able to get those formats if you download without cookies?

**Comment by cheese529** [comment]
> It looks like youtube has won this battle.

I can also confirm the iOS player no longer works once you are hit with this ban, tried everything and nothing fixed the problem until I reset my IP address. I downloaded about maybe 20 videos before I got hit with the ban again.

**Comment by spirillen** [comment]
> It looks like youtube has won this battle.

Alphabet didn't win... they lost, as they no longer have any viewers to their contents creators... and what happens then?

**Comment by aniqueta** [comment]
Given this situation, any advice on how to use ` --cookies` (with an extracted cookies file) to get access to a private playlist (e.g., my watchlist) but then download the videos without sending cookies and triggering this issue? Right now, it's all one yt-dlp command, and I assume yt-dlp is sending cookies for the video downloads as well? But perhaps not, since cookies aren't in the config file? Thanks. (Related to #10108 and #10128 which I believe are duplicate issues.)

**Comment by bashonly** [comment]
@aniqueta I would avoid using cookies entirely and use https://github.com/coletdjnz/yt-dlp-youtube-oauth2 instead.

You could pass auth and print a list of video links from the private playlist with `--flat-playlist --print-to-file webpage_url links.txt`, and then to download them, do not pass auth and feed the list back to yt-dlp with `yt-dlp -a links.txt`

**Comment by Deneb128** [comment]
As @bashonly made me notice, my experience ( https://github.com/yt-dlp/yt-dlp/issues/10046#issuecomment-2235891943 ) was related to this issue (thanks because I had lost this link and I could not find it again this morning).

I want to say that the oauth2 method does not work for me, though. If I put my credentials, it ignores them or, at least, it is useless for member only videos. If I try with oauth2 as username and any random password (even `--password '  '` but not `--password ''`), it gives me the code to enter in the sign in page for an android TV, but it does not work (I guess because, well, I've tried with oauth2 as username)

**Comment by MJBlack9000** [comment]
Slightly off-topic, but the relevant "Sign in to confirm you're not a bot" issues have all been limited to collaborators.
I have a VPS with Hetzner that was blocked immediately. I get the "Sign in..." prompt in yt-dlp and in the browsers.
But! I have a Firefox session in my browser, which is about a month or two old, where even from this same IP I can view the videos.
If I launch a private tab in the same browser - I get the "Sign in..." instead.
I have never logged in from that session, so it's something to do with the existing cookies/localstorage.
If someone is interested in looking into this, I can provide all of that and run some additional tests.

**Comment by bashonly** [comment]
> But! I have a Firefox session in my browser, which is about a month or two old, where even from this same IP I can view the videos.
> I have never logged in from that session, so it's something to do with the existing cookies/localstorage.
> If someone is interested in looking into this, I can provide all of that and run some additional tests.

@MJBlack9000 Could you share the value of that session's `VISITOR_INFO1_LIVE` cookie?

**Comment by MJBlack9000** [comment]
@bashonly I don't think I have it, actually. I could be mistaken, but here is all that my Firefox shows for YouTube
![Cookies](https://github.com/user-attachments/assets/3c7b7ead-b367-47a2-8374-ecc46c6ffc08)

I can share more if needed; but didn't find what you requested in local- or sessionStorage either

**Comment by bashonly** [comment]
@MJBlack9000 if you pass `--cookies-from-browser firefox` to yt-dlp, does that work around the "sign in" error?

**Comment by MJBlack9000** [comment]
@bashonly
TL;DR it doesn't help, still get the same error with yt-dlp.

I've tried just with "firefox", and it loaded a whole bunch of cookies from my FF.
But I am using Waterfox for YT. So I specified the proper profile path `--cookies-from-browser firefox:$HOME/.waterfox/profile.youtube/`; and it loaded a different number of cookies than from default FF profile. But still showed the same "sign in to confirm you're not a bot" error.

I use that profile only for YT, so I can even go as far as to execute some custom script on it, that'll gather the information you might be interested in. If you are inclined to do so. I can also try to find that information for you, if you tell me what to look for.

P.S. apologies for the late reply.

**Comment by MJBlack9000** [comment]
I've got some interesting news.
On my second browser profile, where there was no stored YT session (but where I tried opening some YT videos and left cookies behind), YT now started working. If I go to private browsing - I can still see that my IP is blocked. But it started working in normal mode, where there are YT cookies.

It seems that YT checks the age of your session, while deciding whether to serve you content or not.

**Comment by chuckfluid** [comment]
Currently also experiencing this issue running python code through a heroku server. I was experimenting and ended up making a python code in my server where I could paste the cookie data and run 2 commands and relaunch the heroku server to refresh the cookies. 
method has still not completely worked on the server side just has worked locally. 
I just found this page today so should I keep trying to see if this method might work? Or do you all think that this cookies from browser will end up being a better method? 
keep in mind, the method I’m working on right now still requires a throw away Google account for cookies so you don’t get your own banned. So this definitely doesn’t seem like the ideal method.

**Comment by coletdjnz** [comment]
With #10648 merged, supplying a valid PO Token may help in some cases. 

See the [PO Token Guide](https://github.com/yt-dlp/yt-dlp/wiki/Extractors#po-token-guide) and [`po_token` extractor arg](https://github.com/yt-dlp/yt-dlp?tab=readme-ov-file#youtube) for more details.


Note: requires yt-dlp nightly

**Comment by sahfd4eiugf** [comment]
What about using a VPN?
I am browsing the internet at home anyway using Wireguard. So I installed it on my server, which runs yt-dlp, but I am still getting the bot message, strange?

**Comment by bashonly** [comment]
Closing since there's nothing we can do about the way YT runs their platform.

If you don't want to risk your account being blocked, then don't pass cookies.

If you need to pass cookies and want to lower your risk of being blocked, you could try [passing a PO token](https://github.com/yt-dlp/yt-dlp/wiki/Extractors#po-token-guide)
